### ISSUE TITLE
Adaptation for  create_savedmodel_simple_fcn.py

### ISSUE THREAD

**Issue by resnow92** [remicres/otbtf#21]
Hello, 
I managed to run the create_savedmodel_simple_fcn.py and generate classification with OTB-TF for images with 15 concatenated bands and for 3 classes. I have few questions:

1) first, does it make sens for you to run patch extraction on 3 different images (maybe more) in different zones and then concatenate these patchs and pass them to the tensorflowModelTrain application ?

2) As mentioned in the tutorial, this architecture is basic, and i would like to change it to improve results on my classification (multi-images if it is possible). Do you think it pertinent or this architecture is sufficient ? How to do without knowledge on Tensorflow?

3) Last question, I tried to change the patch size from 16 to 24 but it give me an error with the tensorflowModelTrain ("logits and labels mus have the same first dimension...), which parameter I need to change ? 

Thanks.

**Comment by remicres** [comment]
Well, In short: 

1) yes it makes sense if you want to grain your model over multiple images. You can use the python API of OTB to do that (just concatenate numpy arrays, then give the result as numpy array to OTB)

2) No need to be an expert with TensorFlow, but it would be difficult without a basic understanding of what convolutions do. 

3) See 2! :smiley: There is plenty of great tutorials on the web

If you are at this point, it might be better for you to dive into TensorFlow or Keras (which allos you to export your models as TensoFlow `SavedModels`, and which is easier to use). Keep in mind that the `TensorflowModelTrain` application is great to begin with deep learning, but experts and advanced practitioners would likely prefer some custom made python using directly numpy/tensorflow to train their models, then use `TensorflowModelServe` to deploy them.

**Comment by resnow92** [comment]
Ok, thanks for your help. 
I am not very comfortable with TensorFlow and I have no idea how to train a model from the generated patch :D, that is why i found OTB-TF very very great ! thanks for that ! but I will explore this possibilities to generate my own model. 
Thanks for your help and time.

**Comment by remicres** [comment]
I think that you can adapt the provided models quite easily, once you have understood how the convolution kernel sizes and strides changes the output tensor (in terms of size, and also physical spacing). The goal is to have a model which generates the output with the correct size and physical alignment, given an input patch! After that, take a look at regularization techniques (batchnorm, dropout) for instance, which help models reaching another level.
